### PR TITLE
TS strict types

### PR DESCRIPTION
**Pull request by gugu**
This PR changes a lot - typescript works with "strict": true

### CODE REVIEW

File: src/passport-saml/saml.ts [hunk 1]
```diff
@@ -1333,7 +1359,7 @@ class SAML {
 
     const wrappedKey = [
       '-----BEGIN PRIVATE KEY-----',
-      ...key.match(/.{1,64}/g),
+      ...key.match(/.{1,64}/g)!,
```
Comment (markstos) [inline src/passport-saml/saml.ts:1362]: I looked up what the "!" was for here. It expresses that the value cannot be `null` or `undefined`. 

https://stackoverflow.com/questions/42273853/in-typescript-what-is-the-exclamation-mark-bang-operator-when-dereferenci

**Comment by markstos** [comment]
QA Log
=======

  * Reviewed diff
  * Noticed test coverage remains intact.

**Comment by markstos** [comment]
@gugu is the "draft" status of this PR correct?

File: src/passport-saml/saml.ts (hunk 1 above)
Comment (gugu) [reply src/passport-saml/saml.ts]: yes, JS had a problem - it did not check if key.match(/.{1,64}/) actually finds something and returns a match. Current PR copies this behavior. Here is transpiled version:

image

Fixed in 36263b2

**Comment by gugu** [comment]
Opened #485 instead

**Pull request closed without merging**